ggml.h

```diff
@@ -2449,6 +2456,11 @@ extern "C" {
     typedef void (*ggml_from_float_t)(const float * GGML_RESTRICT x, void  * GGML_RESTRICT y, int64_t k);
     typedef void (*ggml_vec_dot_t)   (int n, float * GGML_RESTRICT s, size_t bs, const void * GGML_RESTRICT x, size_t bx,
                                       const void * GGML_RESTRICT y, size_t by, int nrc);
+    typedef void (*ggml_from_float_to_mat_t)(const float * GGML_RESTRICT x, void  * GGML_RESTRICT y, int64_t k);
+    typedef void (*ggml_gemv_t)      (int n, float * GGML_RESTRICT s, const void * GGML_RESTRICT vx, const void * GGML_RESTRICT vy,
+                                      int nr, int nc, int ith, int nth);
+    typedef void (*ggml_gemm_t)      (int n, float * GGML_RESTRICT s, const void * GGML_RESTRICT vx, const void * GGML_RESTRICT vy,
+                                      int nr, int nc, int ith, int nth);
```

Comment:
The signatures of `ggml_gemv_t` and `ggml_gemm_t` should not have a notion of threads - i.e. `ith` and `nth`. Probably look to extend `type_traits` with `.ncols` similar to `.nrows` (if needed) and compute the correct offsets of `vx` and `vy` before calling them

Reply:
The signatures of `ggml_gemv_t` and `ggml_gemm_t` no longer include the `ith` and `nth` arguments. We added `.ncols` to `type_traits` and computed the correct offsets of `vx` and `vy` before calling the gemv and gemm functions as suggested.